### DISCUSSION TITLE
Ability to pass referrer to storefront cart

### DISCUSSION THREAD

**Discussion by nick-addison-applications** [Shopify/storefront-api-feedback#31] [Ideas & Requests]
Currently there is no way to pass the referrer details to the cart or checkout. 

As mentioned on the [Shopify Community Forum](https://community.shopify.com/c/Shopify-APIs-and-SDKs/Storefront-API-Website-Specify-Referrer-When-Creating-Checkout/td-p/510916).

This makes the analytics report at /admin/reports/sales_by_referrer appear as if all traffic was referred by "direct".

It may be useful if there was a way to add this information when creating or updating a cart with the storefront API (or if there was some way to pass the data automatically).

**Comment by morganmccunn** [comment]
Hey @nick-addison-applications ! We are aware of limitations with reporting breaking down for the storefront API. It's definitely on the radar but we need to evaluate a few options for fixing this in the long-term. I will share an update once we have clarity on our path here! Thanks for flagging.

**Comment by arobbins** [comment]
I would really benefit from this as well.
